src/stream/mod.rs

```diff
@@ -880,7 +882,7 @@ pub trait Stream {
     /// This method is only available when the `use_std` feature of this
     /// library is activated, and it is activated by default.
     #[cfg(feature = "use_std")]
-    fn buffered(self, amt: usize) -> Buffered<Self>
+    fn buffered(self, amt: usize) -> Buffered<FuturesOrdered<<<Self as Stream>::Item as IntoFuture>::Future>, Self>
```

Comment:
Thanks for the updates! Sorry for the delay just got back from vacation.

I think this is a breaking change though, right? Would it be possible to land this without breaking changes? I think it's fine to have a bit of duplication here to preserve these existing type signatures perhaps?